cimr/processor/yamler.py

```diff
@@ -0,0 +1,400 @@
+            if validate_data_type(data_type):
+                self.data_type = data_type
+        except ValueError:
+            logging.error(f' there is no data_type indicated.')
+            sys.exit(1)
+    
+
+    def check_hash(self):
+        """Compare md5 of the downloaded file to the provided value"""
+        if validate_hash(self.downloaded_file, self.hash):
+            logging.info(f' md5sum verified.')
+        else:
+            raise ValueError(' provided md5sum didn\'t match.')
+
+
+    def download(self):
+        """Check if provided weblink to the file exists. 
+        Download if verified.
+        """
+        self.file_link = self.yaml_data['data_file']['location']['url']
+        self.downloaded_file = self.file_link.split('/')[-1]
+
+        self.outdir_root = 'submitted_data/'
+        pathlib.Path(self.outdir_root).mkdir(exist_ok=True)
+        self.outdir = self.outdir_root + str(self.data_type) + '/'
+        pathlib.Path(self.outdir).mkdir(exist_ok=True)
+
+        if verify_weblink(self.file_link):
+            logging.info(f' starting download')
+            download_file(self.file_link, self.outdir)
+            self.hash = self.yaml_data['data_file']['location']['md5']
+            self.downloaded_file = self.outdir + self.downloaded_file
+        else:
+            logging.error(f' file unavailable')
+            sys.exit(1)
+
+
+    def make_archive_dir(self):
+        """Make a temp dir for downloaded files to be archived"""
+        self.archive_dir = 'submitted_data/downloaded_archive/'
+        pathlib.Path(self.archive_dir).mkdir(exist_ok=True)
+
+
+    def extract_bulk(self):
+        """Extract downloaded bulk file"""
+        import tarfile
+
+        if tarfile.is_tarfile(self.downloaded_file):
+            tarred_data = tarfile.open(
+                self.downloaded_file, 
+                mode='r:*'
+            )
+            # 'multiple' option is added for edge cases where multiple
+            # data type files in a tarfile share the same yaml
+            if self.data_type == 'multiple':
+                verify_dir(tarred_data)
+                tarred_data.extractall(path=self.outdir_root)
+            else:
+                for member in tarred_data.getmembers():
+                    if member.isreg():
+                        member.name = os.path.basename(member.name)
+                        tarred_data.extract(member, path=self.outdir)
+
+        else: # raise exception for invalid archive file
+            raise Exception(' invalid archive file for upload_bulk')
+
+        # per pr#12, moving the downloaded archive file to subdir
+        # 'downloaded_archive' to avoid being processed later
+        # ref: https://github.com/greenelab/cimr-d/pull/12  
+        self.make_archive_dir()
+        new_path = self.archive_dir + self.downloaded_file.split('/')[-1]
+        os.rename(
+            self.downloaded_file,
+            self.archive_dir + self.downloaded_file.split('/')[-1]
```

Comment:
This line can be replaced by `new_path`. (My bad, I should have noticed this when working on the last PR.)